Source/Relativity.Test.Helpers/Relativity.Test.Helpers/ArtifactHelpers/Fields.cs

```diff
@@ -6,99 +6,146 @@
-            return CreateField(client, fieldRequest);
-        }
-        public static int CreateField_WholeNumber(IRSAPIClient client, int workspaceID)
-        {
-            var fieldRequest = new FieldRequest(workspaceID, FieldType.WholeNumber);
-            return CreateField(client, fieldRequest);
-        }
-        public static int CreateField_YesNO(IRSAPIClient client, int workspaceID)
-        {
-            var fieldRequest = new FieldRequest(workspaceID, FieldType.YesNo);
-            return CreateField(client, fieldRequest);
-        }
-        public static int CreateField_SingleChoice(IRSAPIClient client, int workspaceID)
-        {
-            var fieldRequest = new FieldRequest(workspaceID, FieldType.SingleChoice);
-            return CreateField(client, fieldRequest);
-        }
-        public static int CreateField_MultipleChoice(IRSAPIClient client, int workspaceID)
-        {
-            var fieldRequest = new FieldRequest(workspaceID, FieldType.MultipleChoice);
-            return CreateField(client, fieldRequest);
-        }
-    }
+	/// <summary>
+	/// 
+	/// Helpers to interact with Fields in Relativity
+	/// 
+	/// </summary>
+	/// 
+	public class Fields
+	{
+
+		public static int GetFieldArtifactID(String fieldname, int workspaceId)
+		{
+			const string routeName = "GetFieldArtifactId";
+
+			var httpRequestHelper = new HttpRequestHelper<FieldArtifactIdRequestModel>();
+
+			var requestModel = new FieldArtifactIdRequestModel
+			{
+				FieldName = fieldname,
+				WorkspaceId = workspaceId
+			};
+
+			var content = httpRequestHelper.GetRequestContent(requestModel);
+			var restAddress = httpRequestHelper.GetRestAddress(routeName);
+
+			FieldArtifactIdResponseModel responseModel;
+			var client = httpRequestHelper.GetClient();
+			using (client)
+			{
+				var response = client.PostAsync(restAddress, content).Result;
+				if (!response.IsSuccessStatusCode)
+				{
+					throw new TestHelpersException("Failed to get field Aritfact ID.");
+				}
+				var responseString = response.Content.ReadAsStringAsync().Result;
+				responseModel = JsonConvert.DeserializeObject<FieldArtifactIdResponseModel>(responseString);
+			}
+
+			return responseModel.ArtifactId;
+		}
+
+		public static int GetFieldCount(int artifactId, int workspaceId)
+		{
+			const string routeName = "GetFieldCount";
+
+			var httpRequestHelper = new HttpRequestHelper<FieldCountRequestModel>();
+
+			var requestModel = new FieldCountRequestModel
+			{
+				FieldArtifactId = artifactId,
+				WorkspaceId = workspaceId
+			};
+
+			var content = httpRequestHelper.GetRequestContent(requestModel);
+			var restAddress = httpRequestHelper.GetRestAddress(routeName);
+
+			FieldCountResponseModel responseModel;
+			var client = httpRequestHelper.GetClient();
+			using (client)
+			{
+				var response = client.PostAsync(restAddress, content).Result;
+				if (!response.IsSuccessStatusCode)
+				{
+					throw new TestHelpersException("Failed to get field count");
```

Comment:
you should include the result message as well. it will be unclear to people consuming this function why the call failed